retry_test.go

```diff
@@ -353,7 +353,37 @@ func TestRetryQueryFail(t *testing.T) {
 	}
 	_, err = newRetryHTTP(context.Background(),
 		client,
-		emptyRequest, urlPtr, make(map[string]string), 15*time.Second, defaultTimeProvider, nil).doPost().setBody([]byte{0}).execute()
+		emptyRequest, urlPtr, make(map[string]string), 15*time.Second, 100, defaultTimeProvider, nil).doPost().setBody([]byte{0}).execute()
+	if err == nil {
```

Comment:
We have an assertion library for such checks :)

Reply:
Changed nil checks in this file to use `assertNilf/assertNotNilf`